Set up API endpoints and shared volume to interact with WEFEgui

paulapreuss opened the pull request:
This is a setup to trigger the WEFEDemand scripts from the GUI and save the results to a shared volume. This PR is the counterpart to [this PR](https://github.com/rl-institut/WEFEgui/pull/14) on WEFEgui.

paulapreuss committed:
Add docker configuration

paulapreuss committed:
Fix typos

paulapreuss committed:
Set up API endpoints

paulapreuss committed:
Update requirements

paulapreuss committed:
Add port and change network name

Bachibouzouk committed:
Fix imports

Bachibouzouk commented:
@paulapreuss - I would need the SURVEY_KEY you used to test this one

Right now I think we should setup celery worker for the tasks, otherwise the app is going to be cluterred

Bachibouzouk commented on demo/app.py:
```diff
@@ -0,0 +1,48 @@
+
+app = Flask(__name__)
+
+def run_script(script, args):
```
I'd setup workers rather than run code with subprocess

Bachibouzouk commented on demo/app.py:
```diff
@@ -0,0 +1,48 @@
+@app.route("/preprocessing", methods=["POST"])
+def run_preprocessing():
+    data = request.get_json()
+    script = "demo/preprocessing_demo.py"
```
modify this file such that there is a fonction to import taking the same arguments

paulapreuss committed:
Call ramp simulation from function instead of CLI

Using the CLI also still works

paulapreuss committed:
Delete preprocessing endpoint and add index

Only ramp simulation endpoint should be necessary, since it already calls preprocessing

paulapreuss committed:
fixup! Call ramp simulation from function instead of CLI

paulapreuss committed:
Switch app from Flask to FastAPI

paulapreuss commented:
Switched to FastAPI because I think it will be better, did not implement workers yet. The dockerfile does not work yet, I think (I didn't do much testing here) but running from `python app.py` from `demo` and running locally might be good enough for demo (`localhost:5000`) is currently hardcoded in `WEFEGui` as the API.
